File: app/src/main/java/kr/com/misemung/realm/repository/AirRepository.java
package kr.com.misemung.realm.repository;




import io.realm.Realm;
import io.realm.RealmResults;
import kr.com.misemung.realm.entity.AirRecord;
import kr.com.misemung.vo.AirInfo;

public class AirRepository {

    public static class Air {
        public static void set(String stationName, AirInfo airInfo) {
            Realm realm = Realm.getDefaultInstance();

            try {
                realm.beginTransaction();

                Number nextID = (realm.where(AirRecord.class).max("id"));
                if (nextID == null) {
                    nextID = 2;
                } else {
                    nextID = nextID.intValue() + 1;
                }

                AirRecord record = new AirRecord(airInfo);
                record.stationName = stationName;
                record.id = (int) nextID;

                realm.insertOrUpdate(record);
                realm.commitTransaction();
            } catch (Exception e) {
                e.printStackTrace();
                realm.cancelTransaction();
            } finally {
                realm.close();
            }
        }

        public static void setCurrent(int id, String stationName, AirInfo airInfo) {
            Realm realm = Realm.getDefaultInstance();

            try {
                realm.beginTransaction();

                AirRecord record = new AirRecord(airInfo);
                record.stationName = stationName;
                record.id = id;

                realm.insertOrUpdate(record);
                realm.commitTransaction();
            } catch (Exception e) {
                e.printStackTrace();
                realm.cancelTransaction();
            } finally {
                realm.close();
            }
        }

        public static Number getId() {
            Realm realm = Realm.getDefaultInstance();

            Number nextID = (realm.where(AirRecord.class).max("id"));
            if (nextID == null) {
                nextID = 2;
            } else {
                nextID = nextID.intValue() + 1;
            }
            return nextID;
        }

        public static RealmResults<AirRecord> selectByAllList() {

            return Realm.getDefaultInstance().where(AirRecord.class)
                    .notEqualTo("id", 1)
                    .findAll();
        }

        public static AirRecord selectByGPSData(int id) {

            return Realm.getDefaultInstance().where(AirRecord.class)
                    .equalTo("id", id)
                    .findFirst();
        }

        public static AirRecord selectByDustData(int id, String stationName) {

            return Realm.getDefaultInstance().where(AirRecord.class)
                    .equalTo("id", id)
                    .equalTo("stationName", stationName)
                    .findFirst();
        }

        public static void updateDustData(int id, AirInfo airInfo, String stationName) {
            Realm realm = Realm.getDefaultInstance();

            try {
                realm.beginTransaction();

                AirRecord record = new AirRecord(airInfo);
                record.id = id;
                record.stationName = stationName;

                realm.insertOrUpdate(record);
                realm.commitTransaction();
            } catch (Exception e) {
                e.printStackTrace();
                realm.cancelTransaction();
            } finally {
                realm.close();
            }
        }

        public static void deleteDustData(int id) {
            Realm realm = Realm.getDefaultInstance();

            realm.executeTransaction(realm1 -> {
                RealmResults<AirRecord> result
                        = realm1.where(AirRecord.class).equalTo("id", id).findAll();
                result.deleteAllFromRealm();
            });
        }

    }
}

File: app/src/main/java/kr/com/misemung/realm/repository/AirRepository.kt
package kr.com.misemung.realm.repository

import io.realm.Realm
import io.realm.RealmResults
import kr.com.misemung.realm.entity.AirRecord
import kr.com.misemung.vo.AirInfo

class AirRepository {
    object Air {
        operator fun set(stationName: String?, airInfo: AirInfo) {
            val realm = Realm.getDefaultInstance()
            try {
                realm.beginTransaction()
                var nextID = realm.where(AirRecord::class.java).max("id")
                nextID = if (nextID == null) {
                    2
                } else {
                    nextID.toInt() + 1
                }
                val record = AirRecord(airInfo)
                record.stationName = stationName
                record.id = nextID
                realm.insertOrUpdate(record)
                realm.commitTransaction()
            } catch (e: Exception) {
                e.printStackTrace()
                realm.cancelTransaction()
            } finally {
                realm.close()
            }
        }

        fun setCurrent(id: Int, stationName: String?, airInfo: AirInfo) {
            val realm = Realm.getDefaultInstance()
            try {
                realm.beginTransaction()
                val record = AirRecord(airInfo)
                record.stationName = stationName
                record.id = id
                realm.insertOrUpdate(record)
                realm.commitTransaction()
            } catch (e: Exception) {
                e.printStackTrace()
                realm.cancelTransaction()
            } finally {
                realm.close()
            }
        }

        val id: Number
            get() {
                val realm = Realm.getDefaultInstance()
                var nextID = realm.where(AirRecord::class.java).max("id")
                nextID = if (nextID == null) {
                    2
                } else {
                    nextID.toInt() + 1
                }
                return nextID
            }

        fun selectByAllList(): RealmResults<AirRecord> {
            return Realm.getDefaultInstance().where(AirRecord::class.java)
                .findAll()
        }

        fun selectByGPSData(id: Int): AirRecord? {
            return Realm.getDefaultInstance().where(AirRecord::class.java)
                .equalTo("id", id)
                .findFirst()
        }

        fun selectByDustData(id: Int, stationName: String?): AirRecord? {
            return Realm.getDefaultInstance().where(AirRecord::class.java)
                .equalTo("id", id)
                .equalTo("stationName", stationName)
                .findFirst()
        }

        fun updateDustData(id: Int, airInfo: AirInfo, stationName: String?) {
            val realm = Realm.getDefaultInstance()
            try {
                realm.beginTransaction()
                val record = AirRecord(airInfo)
                record.id = id
                record.stationName = stationName
                realm.insertOrUpdate(record)
                realm.commitTransaction()
            } catch (e: Exception) {
                e.printStackTrace()
                realm.cancelTransaction()
            } finally {
                realm.close()
            }
        }

        fun deleteDustData(id: Int) {
            val realm = Realm.getDefaultInstance()
            realm.executeTransaction { realm1: Realm ->
                val result = realm1.where(
                    AirRecord::class.java
                ).equalTo("id", id).findAll()
                result.deleteAllFromRealm()
            }
        }
    }
}